tests/unit/test_solvers/test_solution.py

```diff
@@ -371,115 +370,102 @@ def test_plot(self):
-            np.testing.assert_array_equal(solution.data["c + d"], data_load["c_plus_d"])
-
-            # to csv
-            with pytest.raises(
-                ValueError, match="only 0D variables can be saved to csv"
-            ):
-                solution.save_data(f"{test_stub}.csv", to_format="csv")
-            # only save "c" and "2c"
-            solution.save_data(f"{test_stub}.csv", ["c", "2c"], to_format="csv")
-            csv_str = solution.save_data(variables=["c", "2c"], to_format="csv")
-
-            # check string is the same as the file
-            with open(f"{test_stub}.csv") as f:
-                # need to strip \r chars for windows
-                assert csv_str.replace("\r", "") == f.read()
-
-            # read csv
-            df = pd.read_csv(f"{test_stub}.csv")
-            np.testing.assert_allclose(
-                df["c"], solution.data["c"], rtol=1e-7, atol=1e-6
-            )
-            np.testing.assert_allclose(
-                df["2c"], solution.data["2c"], rtol=1e-7, atol=1e-6
-            )
+        data_load = pybamm.load(f"{test_stub}.pickle")
+        np.testing.assert_array_equal(solution.data["c"], data_load["c"])
+        np.testing.assert_array_equal(solution.data["d"], data_load["d"])
 
-            # to json
-            solution.save_data(f"{test_stub}.json", to_format="json")
-            json_str = solution.save_data(to_format="json")
+        # to matlab
+        solution.save_data(f"{test_stub}.mat", to_format="matlab")
+        data_load = loadmat(f"{test_stub}.mat")
+        np.testing.assert_array_equal(solution.data["c"], data_load["c"].flatten())
+        np.testing.assert_array_equal(solution.data["d"], data_load["d"])
 
-            # check string is the same as the file
-            with open(f"{test_stub}.json") as f:
-                # need to strip \r chars for windows
-                assert json_str.replace("\r", "") == f.read()
+        with pytest.raises(ValueError, match="matlab"):
+            solution.save_data(to_format="matlab")
 
-            # check if string has the right values
-            json_data = json.loads(json_str)
-            np.testing.assert_allclose(
-                json_data["c"], solution.data["c"], rtol=1e-7, atol=1e-6
-            )
-            np.testing.assert_allclose(
-                json_data["d"], solution.data["d"], rtol=1e-7, atol=1e-6
-            )
+        # to matlab with bad variables name fails
+        solution.update(["c + d"])
+        with pytest.raises(ValueError, match="Invalid character"):
+            solution.save_data(f"{test_stub}.mat", to_format="matlab")
+        # Works if providing alternative name
+        solution.save_data(
+            f"{test_stub}.mat",
+            to_format="matlab",
+            short_names={"c + d": "c_plus_d"},
+        )
+        data_load = loadmat(f"{test_stub}.mat")
+        np.testing.assert_array_equal(solution.data["c + d"], data_load["c_plus_d"])
```

Comment:
Same as above. These files are not looked at often, and we would appreciate the cleanup :)